staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -579,37 +638,43 @@ func determineResyncPeriod(desired, check time.Duration) time.Duration {
 const minimumResyncPeriod = 1 * time.Second
 
 func (s *sharedIndexInformer) AddEventHandlerWithResyncPeriod(handler ResourceEventHandler, resyncPeriod time.Duration) (ResourceEventHandlerRegistration, error) {
+	return s.AddEventHandlerWithConfig(context.Background(), handler, HandlerConfig{ResyncPeriod: resyncPeriod})
+}
+
+func (s *sharedIndexInformer) AddEventHandlerWithConfig(ctx context.Context, handler ResourceEventHandler, config HandlerConfig) (ResourceEventHandlerRegistration, error) {
```

Comment:
The context is only used to pass down the logger. The rationale for using a context is:
- More natural for some call sites because they have a context and not a logger.
- Future-proof: theoretically we could switch from `logr.Logger` as the main logging API to `slog.Logger` (not planned, but who knows...)
- Callers which only use slog and https://github.com/veqryn/slog-context don't need to deal with klog and logr.

Reply:
> The context is only used to pass down the logger. 

Not anymore, now it supports cancellation: once cancelled, all event processing in the background goroutines stops immediately (https://github.com/kubernetes/kubernetes/pull/126387#discussion_r1811427431).

Reply:
Replaced by mapping cancellation to handler removal.